test/error_messages.coffee

```diff
@@ -1664,25 +1664,26 @@ test 'CSX error: invalid attributes', ->
 
 test '#5034: CSX error: Adjacent JSX elements must be wrapped in an enclosing tag', ->
   assertErrorFormat '''
-    render = ->
+    render = -> (
```

Comment:
I think we should test that it errors out both with and without parens

Reply:
I talked about this a bit earlier.
You could say that, but you could also say that

```coffeescript
->
  <a />
  <b />
```
really is
```coffeescript
->
  return (
    <a />
    <b />
  )
```
just like
```coffeescript
->
  a: 1
  b: 2
```
is
```coffeescript
->
  return {
    a: 1
    b: 2
  }
```
instead of
```coffeescript
->
  {a: 1}
  return {b: 2}
```

Reply:
I don’t think you could say that parentheses are implied. JSX objects on subsequent lines aren’t assumed to be part of a unified construct the way an object is. They don’t compile into a single entity, for one thing, even with parentheses:

```coffee
jsx = (
  <a />
  <b />
)
```
```jsx
jsx = (<a />, <b />);
```

That’s not the way objects behave. The separate keys of an object are unified by virtue of being all part of the same block, as defined by indentation. Adjacent JSX tags are just that, adjacent JSX tags, unless they’re enclosed by a parent tag or fragment.

Reply:
OK. I also want to point out that this:

> I agree it’s likely a mistake on the user’s part, but catching “most of the time this is wrong” errors is the job of a linter, not the compiler.

Could very well apply to the original warning. Babel could compile it to two createElement doing nothing.

Reply:
> Could very well apply to the original warning. Babel could compile it to two createElement doing nothing.

It could, but the JSX spec tells it not to. I think that's the distinction.